src/main/java/com/intuit/graphql/orchestrator/batch/GraphQLServiceBatchLoader.java

```diff
@@ -154,8 +160,25 @@ private CompletableFuture<List<DataFetcherResult<Object>>> load(List<DataFetchin
       }
     }
 
-    final SelectionSet filteredSelection = selectionSetBuilder.build();
+    SelectionSet filteredSelection = selectionSetBuilder.build();
```

Comment:
I would recommend the query optimization logic moved to its own class and use the new class here.